synapse/rest/client/v2_alpha/register.py

```diff
@@ -389,8 +389,13 @@ def on_POST(self, request):
             assert_params_in_dict(params, ["password"])
 
             desired_username = params.get("username", None)
-            new_password = params.get("password", None)
             guest_access_token = params.get("guest_access_token", None)
+            new_password = params.get("password", None)
+
+            if new_password and isinstance(new_password, bytes):
+                # We may not need to decode the password, if it came from the
```

Comment:
this smells wrong. Surely we should be storing the same type in the session as is in the original params?

Reply:
also, haven't `params` already been decoded?

Reply:
The original params are `str` (so bytes on Python 2) because json.loads returns strs, not Unicode, we don't want that.